flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/TableSchemaUtils.java

```diff
@@ -71,4 +72,45 @@ public static TableSchema checkNoGeneratedColumns(TableSchema schema) {
 		}
 		return schema;
 	}
+
+	/**
+	 * Creates a builder with given table schema.
+	 *
+	 * @param oriSchema Original schema
+	 * @return the builder with all the information from the given schema
+	 */
+	public static TableSchema.Builder builderWithGivenSchema(TableSchema oriSchema) {
```

Comment:
Please add test for this method and `copySchemaWithoutConstraint`, to make sure this method is up to date when `TableSchema` changes.